src/main/java/org/jboss/aerogear/test/builders/PushApplicationBuilder.java

```diff
@@ -0,0 +1,96 @@
+package org.jboss.aerogear.test.builders;
+
+import org.apache.commons.beanutils.BeanUtils;
+import org.jboss.aerogear.unifiedpush.api.*;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Builder for push applications
+ */
+public class PushApplicationBuilder {
+
+    /**
+     * Temporary push application object used to store values
+     */
+    private final PushApplication pushApplication;
+
+    private final List<Variant> variants;
+
+    private PushApplicationBuilder(final String id, final String name) {
+        this.pushApplication = new PushApplication();
+        this.pushApplication.setId(id);
+        this.pushApplication.setName(name);
+
+        this.variants = new ArrayList<Variant>();
+    }
+
+    private PushApplicationBuilder(final PushApplication pushApplication) {
+        this.pushApplication = pushApplication;
+        this.variants = pushApplication.getVariants() != null ? pushApplication.getVariants() : new ArrayList<Variant>();
+    }
+
+    public static PushApplicationBuilder forApplication(final String id, final String name) {
+        return new PushApplicationBuilder(id, name);
+    }
+
+    public PushApplicationBuilder withDescription(final String description) {
+        this.pushApplication.setDescription(description);
+        return this;
+    }
+
+    public PushApplicationBuilder withDeveloper(final String developer) {
+        this.pushApplication.setDeveloper(developer);
+        return this;
+    }
+
+    public PushApplicationBuilder withMasterSecret(final String masterSecret) {
+        this.pushApplication.setMasterSecret(masterSecret);
+        return this;
+    }
+
+    public PushApplicationBuilder withPushApplicationID(final String pushApplicationID) {
+        this.pushApplication.setPushApplicationID(pushApplicationID);
+        return this;
+    }
+
+    public PushApplicationBuilder withVariant(final Variant variant) {
+        this.variants.add(variant);
+        return this;
+    }
+
+
+    private Variant cloneVariant(Variant variant) throws Exception {
+        Variant ret = variant.getClass().newInstance();
+
+        BeanUtils.copyProperties(ret, variant);
+        return ret;
+    }
+
+    private List<Variant> cloneVariants(List<Variant> variants) throws Exception {
+        List<Variant> ret = new ArrayList<Variant>();
```

Comment:
Because that would only copy a reference to the variant. I need the whole variant object to be copied (deep copy)